### ISSUE TITLE
Adding order arguments to search queries. 

### ISSUE THREAD

**Issue by dopry** [wagtail/wagtail#9890]
### Is your proposal related to a problem?

I am working on ordering in wagtail grapple. I've had to do a bit of work in grapple to explicitly order queries. It took me a while to find I need to set order_by_relevance to false.  I found it really awkward that I needed to do more than just specify an explicit order on a query I provided as a basis for the search. 

After some digging I figured out that order_by_relevance can't be inferred because Page.objects.all() is ordered by path by default.  That was after trying to order the result get_search_backend().search(...), which doesn't work because it doesn't return a queryset. It returns an implementation of BaseSearchResult, which makes sense given search support elasticsearch. Annotating all a queryset  through a proxy with the scores from ES would undo any performance benefit, so it all makes sense...  

Passing a queryset into search is a way to pre-filter the results that will need to be scored and ranked for the search engine.  Which makes sense.  

### Describe the solution you'd like

I think it would be nice if there were an order argument in place of order_by_relevance that defaults to ['relevance'] 
We should allow any string and just check if the model has the field before adding it to the ultimate order_by an ignore or want on fields that aren't on the model.

**Comment by gasman** [comment]
Thanks for the suggestion @dopry! To summarise, what you're proposing is for:

    MyModel.objects.search("term", order=["some_field", "other_field"])

to have equivalent functionality to:

    MyModel.objects.order_by("some_field", "other_field").search("term", order_by_relevance=False)

Is that correct?

(If we do need a magic string value to pass to `order` indicate ordering by relevance - and I'm not sure that we do, since `order=None` would probably do the job - I'd suggest `"__relevance__"`, to avoid the possibility of clashing with an actual field named `relevance` and keep consistency with Django's use of things like `fields = "__all__"` for ModelForms.)

**Comment by dopry** [comment]
@gasman you got it... although looking at

```python
MyModel.objects.search("term", order=["some_field", "other_field"]) # order would default to __relevance__ in this scenario.
```

my concern is that it precludes `.order_by(...)` altogether. and would require changes to how ordering is setup in all code currently calling order_by_relevance=False.


Another option if it is feasible would be...

```python 
MyMode.objects.order_by("__relevance__", "some_field").search("term") 
```

With this version there would be a synthetic __relevance__ field that is only populated when searching and __relevance__ would need to be explicitly requested by search. With this implementation code with order_by_relevance=false wouldn't not need to be changed and any calls to search that are implicitly expecting order by _relevance_ will need to updated to explicitly request order by relevance.

**Comment by Parthmadan-1** [comment]
I want to work on this issue.. Any updates on it?

**Comment by gasman** [comment]
@dopry My suggestion is for the `order` kwarg on `.search` to essentially act as a shorthand for inserting an `.order_by` clause and setting `order_by_relevance=False`. All existing behaviour when _not_ passing an `order` kwarg would be preserved (namely, ordering by relevance unless `order_by_relevance=False` is passed, in which case it will respect the existing ordering on the queryset), so existing code would continue to work.

I don't think the `MyModel.objects.order_by("__relevance__", "some_field").search("term")` approach is feasible, because once we're inside the `.search` method, all we know is that the current queryset has an ordering defined on it - we don't know whether we got there from

    MyModel.objects.order_by("some_field").search("term")

or from a straightforward

    MyModel.objects.search("term")

where `MyModel` just happens to have `"some_field"` set up as its default ordering in `Meta`. In the latter case, we definitely don't want to depart from the current behaviour of ordering by relevance - which means the only appropriate behaviour (in the absence of any options passed to `search` to indicate otherwise) is to ignore the queryset's ordering, regardless of whether that's an explicit `order_by` or an implicit model-level one.

@Parthmadan-1 If @dopry is on board with this as the way forward, then this should be ready to pick up.

**Comment by Parthmadan-1** [comment]
@gasman great, looking forward to contribute.

**Comment by dopry** [comment]
@gasman sure. the implicit model level search is what I think lead to this in the first place. My only concern is that it may not fix my use case where I'm not actually calling search. I'm using grapple and the paginated_results function that accepts a queryset and some other arguments. I'm extending the behavior of grapple for new types with additional filters and ordering options that are applied to the query set before passing on the to paginated results method.  I think you approach will work, but it will also required updating wagtail-grapple's paginated results function to better handle ordering.

**Comment by gasman** [comment]
@dopry It looks to me like this proposed change will simplify the implementation of `resolve_paginated_queryset` here, since the semantics of passing `order` or not, to either apply the explicit order or use the "natural" default, more or less match up. In pseudocode:

    if search_term:
        # this handles both the order=None (order by relevance) and order=['some_field'] case
        qs = qs.search(search_term, order=order)
    elif order:
        qs = qs.order_by(order)
    else:
        pass  # keep the queryset's built-in ordering

Either way, whether it's an exact fit or not, I think we've established enough potential usefulness here that we're good to go ahead with this enhancement.

**Comment by dopry** [comment]
@gasman I agree. I also think it would be cleaner in the [search documentation](https://docs.wagtail.org/en/stable/topics/search/searching.html#custom-ordering). replacing 
```
>>> EventPage.objects.order_by('date').search("Event", order_by_relevance=False)
```
with 
```
>>> EventPage.objects.search("Event", order=['date'])
```

**Comment by Gua00va** [comment]
Is this issue still open??
I would like to work on it.

**Comment by gasman** [comment]
@Gua00va Please see the discussions above - @Parthmadan-1 has already expressed an interest in this, so best to give them a chance to work on this first.

**Comment by dopry** [comment]
@Parthmadan-1 Do need any additional help to get started working on this issue?
